src/ClientIp.php

```diff
@@ -107,33 +107,70 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
      */
     private function getIp(ServerRequestInterface $request)
     {
-        if ($this->remote) {
-            $ip = file_get_contents('http://ipecho.net/plain');
+        if ($ip = $this->getRemoteIp()) {
+            return $ip;
+        }
 
-            if (self::isValid($ip)) {
+        if ($ip = $this->getProxyIp($request)) {
+            if (!in_array($ip, $this->proxyIps)) {
                 return $ip;
             }
         }
 
-        $ip = null;
-        $server = $request->getServerParams();
-
-        if (!empty($server['REMOTE_ADDR']) && self::isValid($server['REMOTE_ADDR'])) {
-            $ip = $server['REMOTE_ADDR'];
+        if ($ip = $this->getLocalIp($request)) {
+            if (!in_array($ip, $this->proxyIps)) {
+                return $ip;
+            }
```

Comment:
Okay, still a bit confused. Shouldn't the choices be:

1. If remote IP detection is enabled, attempt to return remote IP address.
2. If proxy headers are enabled, attempt to return proxied IP address.
3. If local IP address is found:
   - If proxy IPs are defined, and the local IP is in the proxy list, void the local IP.
   - Otherwise return the local IP.